services/QuillLMS/client/app/bundles/Shared/components/shared/dataTable.tsx

```diff
@@ -259,12 +260,35 @@ export class DataTable extends React.Component<DataTableProps, DataTableState> {
     let className = `${dataTableHeaderClassName} ${header.headerClassName}`
     let style: React.CSSProperties = { width: `${header.width}`, minWidth: `${header.width}`, textAlign: `${this.attributeAlignment(header.attribute)}` as CSS.TextAlignProperty }
     let headerContent = header.name
+    let headerTitle = header.name
+    let headerTooltip = null
+
+    if (header.primaryTitle && header.secondaryTitle) {
+      headerTitle = (
+        <div className="multi-line-header">
+          <p>{header.primaryTitle}</p>
+          <p>{header.secondaryTitle}</p>
+        </div>
+      )
+    }
+
+    if(header.tooltipName && header.tooltipDescription) {
+      headerTooltip = (
+        <Tooltip
+          tooltipText={`<p>${header.tooltipName}<br/><br/>${header.tooltipDescription}</p>`}
+          tooltipTriggerText={<img alt={helpIcon.alt} src={helpIcon.src} />}
+        />
+      )
+    }
+
     if (header.isSortable) {
       const sortDirection = sortAscending ? ascending : descending
       const onClick = () => this.changeSort(header.sortAttribute || header.attribute)
       const sortArrow = [header.attribute, header.sortAttribute].includes(sortAttribute) ? <img alt="arrow" className={`sort-arrow ${sortDirection}`} src={arrowSrc} /> : null
       className+= ' sortable'
-      headerContent = <button className="interactive-wrapper focus-on-light" onClick={onClick} type="button">{header.name}{sortArrow}</button>
+      headerContent = <button className="interactive-wrapper focus-on-light" onClick={onClick} type="button">{headerTitle}{sortArrow}{!!headerTooltip && headerTooltip}</button>
+      // resetting to not render tooltip twice
+      headerTooltip = null
```

Comment:
It think you can make this cleaner (less conditionals, less variable re-sets) by always rendering the tooltip, but making it invisible when you don't want it.